product-backend/src/main/java/org/kainos/ea/api/AuthService.java

```diff
@@ -35,4 +41,40 @@ public int createNewUser(RequestUser input) throws FailedToCreateNewUserExceptio
             throw new FailedToCreateNewUserException();
         }
     }
+
+    public String login(Login ClientCredentials)
+            throws  FailedToGetUserException,
+                    FailedToInsertTokenException,
+                    WrongPasswordException,
+                    WrongEmailException {
+        User userDb = authDao.getUser(ClientCredentials.getEmail());
+        if (userDb == null){
+            throw new WrongEmailException();
+        }
+        String providedPassword = ClientCredentials.getPassword();
+        String hashedPasswordFromDB = userDb.getPassword();
+
+        PasswordHasher passwordHasher = new PasswordHasher();
+        boolean authResult = passwordHasher.authenticateUser(providedPassword, hashedPasswordFromDB);
+
+        if (authResult) {
+            Algorithm algorithm = Algorithm.HMAC256("NOT_HARDCODED_SECRET"); // TODO: change to asynchronous algorithm
+            long now = System.currentTimeMillis();
+            long expiry = now + 3_600_000;
+            String jwtToken = JWT.create()
+                    .withSubject(userDb.getEmail())
+                    .withClaim("user_id", userDb.getId())
+                    .withClaim("user_email", userDb.getEmail())
+                    .withClaim("user_role", userDb.getRole())
+                    .withIssuedAt(new Date(now))
+                    .withExpiresAt(new Date(expiry))
+                    .sign(algorithm);
+
+//            authDao.insertToken(jwtToken, userDb.getEmail(), expiry);
+            return jwtToken;
+        } else {
+            throw new WrongPasswordException();
+        }
```

Comment:
You can invert this if and apply a rule of 'fail fast' 
```suggestion
        if (!authResult) {
            throw new WrongPasswordException();
        }
          Algorithm algorithm = Algorithm.HMAC256("NOT_HARDCODED_SECRET"); // TODO: change to asynchronous algorithm
          long now = System.currentTimeMillis();
          long expiry = now + 3_600_000;
          String jwtToken = JWT.create()
                  .withSubject(userDb.getEmail())
                  .withClaim("user_id", userDb.getId())
                  .withClaim("user_email", userDb.getEmail())
                  .withClaim("user_role", userDb.getRole())
                  .withIssuedAt(new Date(now))
                  .withExpiresAt(new Date(expiry))
                  .sign(algorithm);

          return jwtToken;
```